param/ipython.py

```diff
@@ -364,23 +365,47 @@ def __init__(self, reactive):
 
     def __call__(self):
         from param.depends import depends
-        from param.parameterized import resolve_ref
+        from param.parameterized import Undefined, resolve_ref
         from param.reactive import rx
 
+        handle = None
         if isinstance(self._reactive, rx):
             cb = self._reactive._callback
             @depends(*self._reactive._params, watch=True)
             def update_handle(*args, **kwargs):
-                handle.update(cb())
+                if handle is not None:
+                    handle.update(cb())
         else:
             cb = self._reactive
             @depends(*resolve_ref(cb), watch=True)
             def update_handle(*args, **kwargs):
-                handle.update(cb())
+                if handle is not None:
+                    handle.update(cb())
         try:
-            handle = display(cb(), display_id=uuid.uuid4().hex) # noqa
+            obj = cb()
+            if obj is Undefined:
+                obj = None
+            handle = display(obj, display_id=uuid.uuid4().hex) # noqa
         except TypeError:
             raise NotImplementedError
 
+def async_executor(func):
+    event_loop = None
+    try:
+        ip = get_ipython()  # noqa
+        if ip.kernel:
+            # We are in Jupyter and can piggyback the tornado IOLoop
+            from tornado.ioloop import IOLoop
+            ioloop = IOLoop.current()
+            event_loop = ioloop.asyncio_loop # type: ignore
+    except Exception:
+        pass
+    if event_loop is None:
+        event_loop = asyncio.get_event_loop()
+    if event_loop.is_running():
+        ioloop.add_callback(func)
```

Comment:
Can't this be unbound?

Have you thought about `asyncio.run` with `loop_factory`? This was added in 3.12, so a compatibility layer is needed for earlier versions; see https://github.com/dask/distributed/pull/7969 for inspiration.